src/libraries/System.IO.Pipelines/tests/PipeReaderCopyToAsyncTests.cs

```diff
@@ -86,38 +86,6 @@ public async Task CopyToAsyncPipeWriterWorks()
             targetPipe.Writer.Complete();
         }
 
-        [Fact]
-        public async Task CopyToAsyncPipeWriterResume()
```

Comment:
These tests need to run for both the base implementation and the derived implementation. You can leave these tests here and tests with different PipeReader implementations.

Reply:
One alternative would be to use TheoryData to pass different PipeReader implementations. The Pipe.Reader and `StreamPipeReader`

Reply:
I have reused all the tests via inheritance.
I have removed the CopyToAsyncPipeWriterResume which I added in https://github.com/dotnet/runtime/pull/51147#discussion_r612795026 and which is not testing the expected behavior (see open issue #51272)